### PR TITLE
Initial DynamoDB implementation

### PR DESCRIPTION
**Pull request by underscoreHao**
4346 - Initial DynamoDB implementation

#### 📲 What

- Changes to the domain object needed for DynamoDB
- DI setup for DynamoDB
- Conditional compilation setup for DynamoDB
- Making relevant changes to the DynamoDB repository

#### 🤔 Why

We want to be able to perform CRUD operations to DynamoDB as a data store

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: src/api/xxAMIDOxx.xxSTACKSxx.API.ComponentTests/Fixtures/CreateCategoryFixture.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ protected override void RegisterDependencies(IServiceCollection collection)
         {
             base.RegisterDependencies(collection);
 
-            // Mocked external dependencies, the setup should 
+            // Mocked external dependencies, the setup should
```
Comment (alexanderwjrussell) [inline src/api/xxAMIDOxx.xxSTACKSxx.API.ComponentTests/Fixtures/CreateCategoryFixture.cs:38]: Just a question:

When doing multiline comments would it not be better to use:
`/*
            * <insert here>
            */ `

Comment (underscoreHao) [reply src/api/xxAMIDOxx.xxSTACKSxx.API.ComponentTests/Fixtures/CreateCategoryFixture.cs:38]: No. Multiline comments are _generally_ fine for copyrights/big chunks of texts at the very top of files, but for comments in actual code they're a relic. The single line comments are always better for readability. Also multiline comments have weird nesting issues and are error prone. Also Microsoft are clear in the convention - https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#commenting-conventions

File: src/api/xxAMIDOxx.xxSTACKSxx.Domain/MenuAggregateRoot/Entities/Converters/DynamoDbCategoryConverter.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using xxAMIDOxx.xxSTACKSxx.Domain.Entities;
+
+namespace xxAMIDOxx.xxSTACKSxx.Domain.MenuAggregateRoot.Entities.Converters
+{
+    // IMPORTANT!!!
+    // Since Categories is IReadOnlyCollection<T> we need the converter to work with that interface both during serialization
+    // and deserialization. If Categories was a property using a CONCRETE type like List<T>, then the DynamoSDK would automatically handle
+    // the collection mapping and we could've made a converter for only a single Category
+    // DynamoDB .NET Object Persistence Model Types - https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DotNetSDKHighLevel.html#DotNetDynamoDBContext.SupportedTypes
+    public class DynamoDbCategoryConverter : IPropertyConverter
```
Comment (alexanderwjrussell) [inline src/api/xxAMIDOxx.xxSTACKSxx.Domain/MenuAggregateRoot/Entities/Converters/DynamoDbCategoryConverter.cs:16]: Is this class something directly from AWS or would it be worth wrapping with some unit tests?

File: src/api/xxAMIDOxx.xxSTACKSxx.API.ComponentTests/Fixtures/CreateCategoryFixture.cs (hunk 1 above)
Comment (alexanderwjrussell) [reply src/api/xxAMIDOxx.xxSTACKSxx.API.ComponentTests/Fixtures/CreateCategoryFixture.cs:38]: Ah okay. Thanks. I hadn't seen that. I will update myself then :)

File: src/api/xxAMIDOxx.xxSTACKSxx.API/Controllers/Menu/GetMenuByIdController.cs [hunk 3]
```diff
@@ -51,12 +51,12 @@ public async Task<IActionResult> GetMenu([FromRoute][Required]Guid id)
                 Id = result.Id,
                 Name = result.Name,
                 Description = result.Description,
-                Categories = result.Categories.Select(i => new Category()
+                Categories = result.Categories?.Select(i => new Category()
```
[the remark this answers is not in the document]
Comment (underscoreHao) [reply src/api/xxAMIDOxx.xxSTACKSxx.API/Controllers/Menu/GetMenuByIdController.cs:54]: No, but what happens in this controller is something I'd like to change in a future story. That mapping shouldn't happen in the controller.

[the remark this answers is not in the document]
Comment (alexanderwjrussell) [reply src/api/xxAMIDOxx.xxSTACKSxx.API/Controllers/Menu/GetMenuByIdController.cs:54]: Very true. Maybe we can refactor in a later sprint.

**Pull request merged**